c_src/scenic/ops/cairo_script_ops.c

```diff
@@ -0,0 +1,663 @@
+#include <cairo.h>
+#include <math.h>
+
+#include "cairo_ctx.h"
+#include "comms.h"
+#include "font.h"
+#include "font_ops.h"
+#include "image.h"
+#include "script.h"
+#include "script_ops.h"
+
+extern device_opts_t g_opts;
+
+static const char* log_prefix = "cairo";
+
+void do_fill_stroke(cairo_t* cr, bool fill, bool stroke)
+{
+  if (fill && stroke)
+    cairo_fill_preserve(cr);
+  else if (fill)
+    cairo_fill(cr);
+
+  if (stroke) cairo_stroke(cr);
+}
+
+void script_ops_draw_line(void* v_ctx,
+                          coordinates_t a,
+                          coordinates_t b,
+                          bool stroke)
+{
+  if (g_opts.debug_mode) {
+    log_script_ops_draw_line(log_prefix, __func__, log_level_info,
+                             a, b, stroke);
+  }
+
+  CAIROcontext_t* p_ctx = (CAIROcontext_t*)v_ctx;
+  cairo_move_to(p_ctx->cr, a.x, a.y);
+  cairo_line_to(p_ctx->cr, b.x, b.y);
+
+  if (stroke) cairo_stroke(p_ctx->cr);
```

Comment:
(Optional) Formatting seems a little inconsistent in places where single statement ifs occur. Some are wrapped in curly braces some aren't. Considering that formatting is inconsistent in this project, this isn't a big deal.